caffeinated/admin/extend/registration_form/Extend_Registration_Form_Admin_Page.core.php

```diff
@@ -1182,6 +1191,73 @@ protected function _email_validation_settings_form()
     }
 
 
+    /**
+     * @throws EE_Error
+     * @throws InvalidArgumentException
+     * @throws InvalidDataTypeException
+     * @throws InvalidInterfaceException
+     */
+    public function privacyConsentFormContent()
+    {
+        echo $this->getPrivacyConsentForm()->get_html_and_js();
```

Comment:
ideally I would prefer that you created a `PrivacyConsentForm` class that was its own class that extended `EE_Form_Section_Proper` and injected `EE_Registration_Config` into its constructor
OR
have a `PrivacyConsentFormHandler` class that extends `\EventEspresso\core\libraries\form_sections\form_handlers\FormHandler` which would negate the need for a lot of the boilerplate logic you currently have in `updatePrivacyConsentCheckboxSettingsForm()`,
and then instead of using the `getPrivacyConsentForm()` getter here, you would use the Loader to retrieve an instance of `PrivacyConsentFormHandler` 

for example:

if  `PrivacyConsentFormHandler`  extended `EventEspresso\core\libraries\form_sections\form_handlers\FormHandler`, then `privacyConsentFormContent()` could maybe change to something like:

```php
public function privacyConsentFormContent()
{
    /** @var \EventEspresso\core\libraries\form_sections\form_handlers\FormHandler $privacy_consent_form_handler*/
    $privacy_consent_form_handler= LoaderFactory::getLoader()->getShared('F/Q/C/N/PrivacyConsentFormHandler');
    echo $formHandler->display();
}
```

and `updatePrivacyConsentCheckboxSettingsForm()` could maybe change to something like:

```php
public function updatePrivacyConsentCheckboxSettingsForm(EE_Registration_Config $reg_config)
{
    /** @var \EventEspresso\core\libraries\form_sections\form_handlers\FormHandler $privacy_consent_form_handler */
    $privacy_consent_form_handler = LoaderFactory::getLoader()->getShared('F/Q/C/N/PrivacyConsentFormHandler');
    $privacy_consent_form_handler->process($reg_config);
    return $reg_config;
}
```

where the `PrivacyConsentFormHandler::generate()` method returned the actual `EE_Form_Section_Proper` object and `PrivacyConsentFormHandler::process()` method did something like:

```php
public function process(EE_Registration_Config $reg_config)
{
    try {
        $valid_data = (array) parent::process();
        $reg_config->setConsentCheckboxEnabled($valid_data['enable'] === 'enable-privacy-consent');
        $reg_config->setConsentCheckboxLabelText(
            $valid_data['enable-privacy-consent']['consent_assertion']
        );
    } catch (EE_Error $e) {
        // handle error
    }
    return $reg_config;
}
```

The benefit of this is that it keeps our classes simpler and focused solely on one responsibility.
It also makes it easier down the road to inject the `PrivacyConsentFormHandler` into this controller class when this route is called and not even have to bother with the Loader.

ie: one day it could be something as simple as:

```php
public function privacyConsentFormContent()
{
    echo $this->form_handler->display();
}

public function updatePrivacyConsentCheckboxSettingsForm(EE_Registration_Config $reg_config)
{
    return  $this->form_handler->process($reg_config);
}
```

Reply:
yeah that sounds like a good suggestion. The refactoring should really just affect the one file so won't be too bad I think